packages/elements/src/SafeAreaProviderCompat.tsx

```diff
@@ -23,19 +24,23 @@ const initialMetrics =
       }
     : initialWindowMetrics;
 
-export default function SafeAreaProviderCompat({ children }: Props) {
+export default function SafeAreaProviderCompat({ children, style }: Props) {
   return (
     <SafeAreaInsetsContext.Consumer>
       {(insets) => {
         if (insets) {
           // If we already have insets, don't wrap the stack in another safe area provider
           // This avoids an issue with updates at the cost of potentially incorrect values
           // https://github.com/react-navigation/react-navigation/issues/174
-          return children;
+          return style == null ? (
+            children
+          ) : (
+            <View style={style}>{children}</View>
+          );
```

Comment:
Conditionally changing whether it's wrapped in a `View` based on a prop is problematic. Changing the `style` prop would cause any local state to be reset due to remount. This needs to be always wrapped in a `View`